src/context/toast.tsx

```diff
@@ -53,6 +54,14 @@ const ToastProvider: React.FC = ({ children }) => {
       addToast({ variant: Variant.Note, message, closable }),
     hide: () => hideToast(),
   };
+
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      hideToast();
+    }, TOAST_TIMEOUT);
+    return () => clearTimeout(timeout);
```

Comment:
I think this should actually be () => { clearTimeout(timeout); hideToast() } since clearTimeout will actually prevent the setTimeout callback from being called.

Reply:
The return statement in a `useEffect` will get run when the component gets unmounted (In this case the Toast will also be umounted since it lives within the component) . The `clearTimeout()` is actually used to avoid memory leaks so we dont want the function in the timeout to run. 
https://www.alexhughes.dev/blog/setTimeout-with-hooks/